### PR TITLE
feat(ui): Add session chart to alert creation page

### PR DESCRIPTION
**Pull request by matejminar**
image

Adding session chart on top of crash rate alert creation page.

Things I still need to address in follow-up PRs:
- format tooltip and yAxis labels
- scale the chart (I ran into some problems with threshold rectangles not aligning properly when chart scaling is turned on)

### CODE REVIEW

File: static/app/views/alerts/incidentRules/triggers/chart/index.tsx [hunk 1]
```diff
@@ -130,11 +155,18 @@ type State = {
...
+    sessionsReloading: false,
   };
 
   componentDidMount() {
-    this.fetchTotalCount();
+    if (this.isSessionAggregate) {
+      this.fetchSessionTimeSeries();
+    } else {
+      this.fetchTotalCount();
+    }
```
Comment (priscilawebdev) [inline static/app/views/alerts/incidentRules/triggers/chart/index.tsx:169]:
```suggestion
      if (this.isSessionAggregate) {
        this.fetchSessionTimeSeries();
        return;
      }

      this.fetchTotalCount();
```

File: static/app/views/alerts/incidentRules/triggers/chart/index.tsx [hunk 2]
```diff
@@ -146,7 +178,11 @@ class TriggersChart extends React.PureComponent<Props, State> {
       prevProps.timeWindow !== timeWindow ||
       prevState.statsPeriod !== statsPeriod
     ) {
-      this.fetchTotalCount();
+      if (this.isSessionAggregate) {
+        this.fetchSessionTimeSeries();
+      } else {
+        this.fetchTotalCount();
+      }
```
Comment (priscilawebdev) [inline static/app/views/alerts/incidentRules/triggers/chart/index.tsx:185]:
```suggestion
      if (this.isSessionAggregate) {
        this.fetchSessionTimeSeries();
        return;
      } 
      
       this.fetchTotalCount();
```

File: static/app/views/alerts/incidentRules/triggers/chart/index.tsx [hunk 3]
```diff
@@ -164,43 +200,147 @@ class TriggersChart extends React.PureComponent<Props, State> {
...
+            maxValue={maxValue}
+            data={data}
+            triggers={triggers}
+            resolveThreshold={resolveThreshold}
+            thresholdType={thresholdType}
+          />
+        )}
+        <ChartControls>
+          <InlineContainer>
+            <React.Fragment>
```
Comment (priscilawebdev) [inline static/app/views/alerts/incidentRules/triggers/chart/index.tsx:308]: do you need a fragment here?

File: static/app/views/alerts/incidentRules/triggers/chart/index.tsx [hunk 4]
```diff
@@ -164,43 +200,147 @@ class TriggersChart extends React.PureComponent<Props, State> {
...
+              <SectionHeading>{t('Total Events')}</SectionHeading>
+              {totalCount !== null ? (
+                <SectionValue>{totalCount.toLocaleString()}</SectionValue>
+              ) : (
+                <SectionValue>&mdash;</SectionValue>
+              )}
```
Comment (priscilawebdev) [inline static/app/views/alerts/incidentRules/triggers/chart/index.tsx:314]:
```suggestion
         <SectionValue>
                {totalCount !== null ? totalCount.toLocaleString() : '\u2014'}
              </SectionValue>
```

**Comment by priscilawebdev** [comment]
can you please post a screenshot? :)

File: static/app/views/alerts/incidentRules/triggers/chart/index.tsx (hunk 3 above)
Comment (matejminar) [reply static/app/views/alerts/incidentRules/triggers/chart/index.tsx:308]: I guess not, just moved that block of code 🤷‍♂️
Will try to remove it.

**Review by priscilawebdev — APPROVED** [review]
lgtm

**Pull request merged**